src/js/util.js

```diff
@@ -476,11 +476,22 @@ var Util;
                 return;
             }
 
-            var list = element.parentElement;
+            var parent = element.parentElement;
 
-            if (list.parentElement.nodeName.toLowerCase() === 'p') { // yes we need to clean up
-                this.unwrap(list.parentElement, ownerDocument);
+            if ((parent.nodeName.toLowerCase() === 'ul' || parent.nodeName.toLowerCase() === 'ol') && (parent.parentElement.nodeName.toLowerCase() === 'ul' || parent.parentElement.nodeName.toLowerCase() === 'ol')) { // yes, we need to fix list nesting (valid is: ul/ol > li > ul/ol )
```

Comment:
Could you move comments at the end of each `if` just before the `if` instead ? It'll be more easy to read